### DISCUSSION TITLE
Problem while executing setup.sh

### DISCUSSION THREAD

**Discussion by Aditya-Sakpal** [agama-project/agama#876] [General]
**I am getting a bunch of errors while running the ./setup.sh :-**
./setup.sh: line 16: [: !=: unary operator expected
+ set -eu
+++ dirname /cygdrive/d/open_source_contri/agama/agama/setup-service.sh
++ realpath /cygdrive/d/open_source_contri/agama/agama
+ MYDIR=/cygdrive/d/open_source_contri/agama/agama
++ id --user
+ '[' 197609 '!=' 0 ']'
+ SUDO=sudo
++ sudo id --user
197609
+ '[' '!=' 0 ']'
/cygdrive/d/open_source_contri/agama/agama/setup-service.sh: line 20: [: !=: unary operator expected
+ test -f /etc/zypp/repos.d/d_l_python.repo
+ sudo apt-cyg --non-interactive addrepo https://download.opensuse.org/repositories/devel:/languages:/python/openSUSE_Tumbleweed/ d_l_python
+ sudo apt-cyg --non-interactive --gpg-auto-import-keys install gcc gcc-c++ make openssl-devel ruby-devel python-langtable-data git augeas-devel jemalloc-devel awk
+ which cargo
which: no cargo in 
+ sudo apt-cyg --non-interactive install cargo
+ cd /cygdrive/d/open_source_contri/agama/agama/service
+ bundle config set --local path vendor/bundle
/cygdrive/d/ope

**I am using Windows 11 and cygwin . I have replaced zypper with apt-cyg in the setup.sh and setup-service.sh** 

**Request to the mentors**:-Please review my problem and provide an appropriate solution , thanks :).

**Comment by imobachgs** [comment]
Hi!

The `setup.sh` and `setup-service.sh` script are expected to run in openSUSE Tumbleweed systems. I would recommend using a virtual machine for this. Alternatively, you could check what the script does and replicate it in your system, but I do not think it is as simple as using `apt-cyg`.

Please, let me know if you have further questions.

Regards,
Imo

**Reply by imobachgs** [reply]
Truth be told, we should be more specific in the documentation to avoid further confusion.
